plugins/modules/nd_setup.py

```diff
@@ -0,0 +1,358 @@
+author:
+- Shreyas Srish (@shrsr)
+options:
+  cluster_name:
+    description:
+    - The name of the ND cluster.
+    type: str
+  ntp_server:
+    description:
+    - The IP address of the NTP server.
+    type: list
+    elements: str
+    aliases: [ ntp_servers ]
+  dns_server:
+    description:
+    - The IP address of the DNS server.
+    type: list
+    elements: str
+    aliases: [ dns_servers ]
+  proxy_server:
+    description:
+    - The proxy server.
+    type: str
+  proxy_username:
+    description:
+    - The username of the proxy server.
+    type: str
+  proxy_password:
+    description:
+    - The password of the proxy server.
+    type: str
+  ignore_proxy:
+    description:
+    - Proxy is ignored for the list of host addresses provided in this option.
+    type: list
+    elements: str
+  dns_search_domain:
+    description:
+    - The DNS search domains.
+    type: list
+    elements: str
+    aliases: [ dns_search_domains ]
+  app_network:
+    description:
+    - The app subnet.
+    - This is a 'pod-to-pod' network and is used to run inter pod traffic.
+    type: str
+  service_network:
+    description:
+    - The service subnet.
+    - This is a virtual level subnet used for communication within the cluster.
+    type: str
+  nodes:
+    description:
+    - The node details to set up Nexus Dashboard and bring up the User Interface.
+    type: list
+    elements: dict
+    suboptions:
+      hostname:
+        description:
+        - The host name of the node.
+        type: str
+        required: true
+      serial_number:
+        description:
+        - The serial number of the node.
+        type: str
+        required: true
+      management_ip_address:
+        description:
+        - The management IP address of the node.
+        type: str
+        required: true
+      username:
+        description:
+        - The username of the node.
+        type: str
+        required: true
+      password:
+        description:
+        - The password of the node.
+        type: str
+        required: true
+      management_network:
+        description:
+        - The network that is used for DNS and NTP communication.
+        type: dict
+        required: true
+        suboptions:
+          ipv4_address:
+            description:
+            - The IPv4 address of the management network.
+            type: str
+            required: true
+          ipv4_gateway:
+            description:
+            - The IPv4 gateway of the management network.
+            type: str
+            required: true
+          ipv6_address:
+            description:
+            - The IPv6 address of the management network.
+            type: str
+          ipv6_gateway:
+            description:
+            - The IPv6 gateway of the management network.
+            type: str
+      data_network:
+        description:
+        - The network that is used for for clustering and communication between sites and applications.
```

Comment:
remove the extra "for" in the description of data_network attribute.

Reply:
Thanks for catching this.
Done.